fn/slice.go

```diff
@@ -65,6 +65,34 @@ func Filter[A any](pred Pred[A], s []A) []A {
 	return res
 }
 
+// FilterMap takes a function argument that optionally produces a value and
+// returns a slice of the 'Some' return values.
+func FilterMap[A, B any](as []A, f func(A) Option[B]) []B {
+	var bs []B
```

Comment:
This may lead to gross over allocation which I think is the wrong choice. If the predicate passes very rarely then the size of the return value will be far under capacity. CPU is cheaper than memory